metasrv/tests/it/api/http/cluster_state_test.rs

```diff
@@ -0,0 +1,175 @@
+                .uri(Uri::from_static("/cluster/state"))
+                .method(Method::GET)
+                .finish(),
+        )
+        .await;
+
+    assert_eq!(response.status(), StatusCode::OK);
+
+    let body = response.into_body().into_vec().await.unwrap();
+    let state =
+        serde_json::from_str::<serde_json::Value>((&String::from_utf8_lossy(&body)).as_ref())?;
+    let voters = state["voters"].as_array().unwrap();
+    let non_voters = state["non_voters"].as_array().unwrap();
+    assert_eq!(voters.len(), 2);
+    assert_eq!(non_voters.len(), 0);
+    meta_node.stop().await?;
+    meta_node1.stop().await?;
+    Ok(())
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_http_service_cluster_state() -> common_exception::Result<()> {
+    let addr_str = "127.0.0.1:0";
+
+    let (_log_guards, ut_span) = init_meta_ut!();
+    let _ent = ut_span.enter();
+
+    let tc0 = new_metasrv_test_context(0);
+    let mut tc1 = new_metasrv_test_context(1);
+
+    tc1.config.raft_config.single = false;
+    tc1.config.raft_config.join = vec![tc0.config.raft_config.raft_api_addr()];
+    tc1.config.admin_api_address = addr_str.to_owned();
+    tc1.config.admin_tls_server_key = TEST_SERVER_KEY.to_owned();
+    tc1.config.admin_tls_server_cert = TEST_SERVER_CERT.to_owned();
+
+    let _ = MetaNode::start(&tc0.config.raft_config).await?;
+    let meta_node1 = MetaNode::start(&tc1.config.raft_config).await?;
+
+    let mut srv = HttpService::create(tc1.config, meta_node1);
+
+    // test cert is issued for "localhost"
+    let state_url = format!("https://{}:0/v1/cluster/state", TEST_CN_NAME);
+    let node_url = format!("https://{}:0/v1/cluster/nodes", TEST_CN_NAME);
+
+    // load cert
+    let mut buf = Vec::new();
+    File::open(TEST_CA_CERT)?.read_to_end(&mut buf)?;
+    let cert = reqwest::Certificate::from_pem(&buf).unwrap();
+
+    tokio::spawn(async move {
+        srv.start().await.expect("HTTP: admin api error");
+        // kick off
+        let client = reqwest::Client::builder()
+            .add_root_certificate(cert)
+            .build()
+            .unwrap();
+        let resp = client.get(state_url).send().await;
+        assert!(resp.is_ok());
+        let resp = resp.unwrap();
+        assert!(resp.status().is_success());
+        assert_eq!("/v1/cluster/state", resp.url().path());
+        let state_json = resp.json::<serde_json::Value>().await.unwrap();
+        assert_eq!(state_json["voters"].as_array().unwrap().len(), 2);
+        assert_eq!(state_json["non_voters"].as_array().unwrap().len(), 0);
+
+        let resp_nodes = client.get(node_url).send().await;
+        assert!(resp_nodes.is_ok());
+        let resp_nodes = resp_nodes.unwrap();
+        assert!(resp_nodes.status().is_success());
+        assert_eq!("/v1/cluster/nodes", resp_nodes.url().path());
+        let result = resp_nodes.json::<Vec<Node>>().await.unwrap();
+        assert_eq!(result.len(), 2);
+    });
+    Ok(())
```

Comment:
A `task:spawn` will just be dropped without executing it at all.

Adding a `panic!()` in it you will see it still passes.

Reply:
I finally figured out why the test passed with `tokio::spawn`, thanks. Updated.